ytmusicapi/mixins/browsing.py

```diff
@@ -836,34 +882,180 @@ def get_song_related(self, browseId: str):
+        :return: Dictionary with song lyrics or `None`, if no lyrics are found. 
+            The `hasTimestamps`-key determines the format of the data.
+
+        
+            Example when `timestamps` is set to `False`, or not timestamps are available::
+
+                {
+                    "lyrics": "Today is gonna be the day\\nThat they're gonna throw it back to you\\n",
+                    "source": "Source: LyricFind",
+                    "hasTimestamps": False
+                }
+            
+            Example when `timestamps` is set to `True` and timestamps are available::
+
+                {
+                    "lyrics": [
+                        LyricLine(
+                            text="I was a liar",
+                            start_time=9200,
+                            end_time=10630,
+                            id=1
+                        ),
+                        LyricLine(
+                            text="I gave in to the fire",
+                            start_time=10680,
+                            end_time=12540,
+                            id=2
+                        ),
+                    ],
+                    "source": "Source: LyricFind",
+                    "hasTimestamps": True
+                }
+
+        """
+
+        lyrics: dict = {}
         if not browseId:
-            raise YTMusicUserError("Invalid browseId provided. This song might not have lyrics.")
+            raise YTMusicUserError(
+                "Invalid browseId provided. This song might not have lyrics.")
+
+        if timestamps:
+            # change the client to get lyrics with timestamps (mobile only)
+            copied_context_client = self.context["context"]["client"].copy()
+            self.context["context"]["client"].update({
+                "clientName": "ANDROID_MUSIC",
+                "clientVersion": "7.21.50"
+            })
 
         response = self._send_request("browse", {"browseId": browseId})
-        lyrics["lyrics"] = nav(
-            response, ["contents", *SECTION_LIST_ITEM, *DESCRIPTION_SHELF, *DESCRIPTION], True
-        )
-        lyrics["source"] = nav(
-            response, ["contents", *SECTION_LIST_ITEM, *DESCRIPTION_SHELF, "footer", *RUN_TEXT], True
-        )
 
-        return lyrics
+        if timestamps:
+            # restore the old context
+            self.context["context"]["client"] = copied_context_client  # type: ignore
+
+        # unpack the response
+
+        # we got lyrics with timestamps
+        if timestamps and (data := nav(response, TIMESTAMPED_LYRICS, True)) is not None:
+            assert isinstance(data, dict)
+
+            if not "timedLyricsData" in data:
+                return None
+
+            lyrics["lyrics"] = list(map(LyricLine.from_raw, data["timedLyricsData"]))
+            lyrics["source"] = data.get("sourceMessage")
+            lyrics["hasTimestamps"] = True
```

Comment:
You could simplify this like so:

```
lyrics["hasTimestamps"] = timestamps and (data := nav(response, TIMESTAMPED_LYRICS, True))
if lyrics["hasTimestamps"]:
...
else:
...
```
Saves the explicit assignments.

Reply:
imo this is worse, because you assign to a dict, just to access it again. 🤷

Reply:
ok, fine with me